Userland/Libraries/LibCrypto/ASN1/ASN1.cpp

```diff
@@ -84,26 +84,25 @@ Optional<Core::DateTime> parse_utc_time(StringView time)
     auto minute = lexer.consume(2).to_uint();
     Optional<unsigned> seconds, offset_hours, offset_minutes;
     [[maybe_unused]] bool negative_offset = false;
-    if (!lexer.next_is('Z')) {
-        if (!lexer.next_is(is_any_of("+-"sv))) {
-            seconds = lexer.consume(2).to_uint();
-            if (!seconds.has_value()) {
-                return {};
-            }
+
+    if (lexer.next_is(is_any_of("0123456789"sv))) {
+        seconds = lexer.consume(2).to_uint();
```

Comment:
This looks like an unchecked assumption about the length again, `yyMMDDHHmms`
Or does consume(n) consume at most n bytes?

Reply:
It's "at most n bytes". I am aware that this means for the `generalized` variant that something like `yyyyMMDDH` (i.e. only a single hour digit) will be accepted as valid, but we don't really worry about that, here.

Reply:
sure, sgtm in that case.